internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -34,6 +34,11 @@ type TxnOp[R any] struct {
 	elseOps    []Op
 }
 
+// txnInterface is a marker interface for generic type TxnOp.
+type txnInterface interface {
+	txn()
+}
+
```

Comment:
You can check specific type, for example `v, ok := value.(*TxnOp[NoResult])`, but not all variants `v, ok := value.(*TxnOp[*])`, therefore I added the `txnInterface` - a marker interface with private method (so it can only be fulfilled from the same package).